src/components/editor/index.js

```diff
@@ -120,38 +131,59 @@ class Draw extends Component {
     this.setState({showActions: false});
   };
 
+  _onToggleLayer = layer => {
+    const {selectedFeature} = this.props.editor;
+    if (!selectedFeature) {
+      return;
+    }
+
+    this.props.onTogglePolygonFilter(layer, selectedFeature.id);
+  };
+
   render() {
     const {
       className,
       clickRadius,
       datasets,
       editor,
-      features,
       layers,
+      filters,
+      onUpdate,
       style
     } = this.props;
+
     const {selectedFeature = {}} = editor;
+    const {lastPosition, showActions} = this.state;
+    const selectedFeatureId = (selectedFeature || {}).id;
```

Comment:
`{selectedFeature = {}} = editor` is doing the same thing. No need to do it again here.

Reply:
that's true